### PR TITLE
Mimic the message reduction approach implemented by the new monorepo

### PR DESCRIPTION
**Pull request by hughdickinson**
Staging branch URL: https://pr-5478.pfe-preview.zooniverse.org

Reduces multiple identical feedback messages to a single message followed by a parenthesized count of the form:
> (_N_ matches)

if _N_ (the number of repeats of the same message) is more than 1 or
> (1 match)

if there is only one occurrence of a unique match.

Only change is in the `getFeedbackMessages` function in `app/features/feedback/classifier/helpers/open-feedback-modal.jsx`

Currently the reduction mode is always active. There is no lab functionality to toggle reduction on/off.

The functionality is demonstrated by this project:
`projects/hughdickinson/galaxy-zoo-clump-scout/classify?workflow=11223`

Add `&env=production` if testing on staging.

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [x] Does talk load correctly?
- [x] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by hughdickinson** [comment]
Tests added and passing when tested locally.

**Comment by srallen** [comment]
@hughdickinson thanks for adding tests! I'll review those. We're still trying to determine if Space Warps's setup is buggy or just configuration and whether or not the changes in this branch have an effect.

File: app/features/feedback/classifier/helpers/get-feedback-messages.spec.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const MULTI_TRUE_MULTI_FALSE_RULES = MULTI_TRUE_MULTI_FALSE.map(generateMinimalRule);
+const MULTI_TRUE_SINGLE_FALSE_RULES = MULTI_TRUE_SINGLE_FALSE.map(generateMinimalRule);
+const SINGLE_TRUE_MULTI_FALSE_RULES = SINGLE_TRUE_MULTI_FALSE.map(generateMinimalRule);
+
+describe('helpers > open-feedback-modal > getFeedbackMessages', function() {
+  it('should exist', function() {
+    expect(getFeedbackMessages).to.be.a('function')
+  })
+
+  it('should reduce an array of messages to a collection of unique messages with counts. ' +
```
Comment (srallen) [inline app/features/feedback/classifier/helpers/get-feedback-messages.spec.js]: Just an organizational suggestion, but you can put this repeated message into another describe block (describe blocks can be nested), so:
```
describe('when reducing an array of messages to a collection of unique messages with counts', function ()
 {
   it('expectation description', function () {

   })
})
```

File: app/features/feedback/classifier/helpers/open-feedback-modal.spec.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    it('The counts for the success message class should be > 1. ' +
+      'The count for the failure message class should be 1.',
+      function() {
+        const reduced = getFeedbackMessages(MULTI_TRUE_SINGLE_FALSE_RULES)
+        expect(reduced).to.deep.equal(['hit (3 matches)', 'miss (1 match)'])
+      })
+
+    it('The counts for the success message class should be > 1. ' +
+      'The count for the failure message class should be 1.',
```
Comment (srallen) [inline app/features/feedback/classifier/helpers/open-feedback-modal.spec.js]: The description here is inaccurate. I believe it should be success is 1 and failure is > 1

Comment (hughdickinson) [reply app/features/feedback/classifier/helpers/open-feedback-modal.spec.js]: Apologies. I missed this. Should now be fixed.

**Comment by srallen** [comment]
Oops! I forgot to approve and merge. Doing that now!

**Pull request merged**